src/layouts/default.layout.astro

```diff
@@ -40,3 +40,11 @@ const { lang = 'en', title } = Astro.props;
     </main>
   </body>
 </html>
+
+<script>
+  const savedTheme = localStorage.getItem('theme');
```

Comment:
This maybe is not the right place to initialize a preselected theme.

Reply:
Why not?